src/proposer/block_builder.cpp

```diff
@@ -114,7 +114,7 @@ class BlockBuilderImpl : public BlockBuilder {
     {
       CScript script_sig = CScript() << CScriptNum::serialize(eligible_coin.target_height)
                                      << ToByteVector(snapshot_hash);
-      tx.vin.emplace_back(uint256(), 0, script_sig);
+      tx.vin.emplace_back(COutPoint(), script_sig);
```

Comment:
Default constructor of `COutPoint` sets `n = (uint32_t) -1`. Maybe we should have a unit test which checks meta input.

Reply:
If only we merged #1098 before

Reply:
It should actually be validated now that I think about it. Ad that should be unit tested

Reply:
Are you then gonna add a test to this PR? @scravy

Reply:
@Gnappuraz @Nizametdinov I added the validation check in this PR ( https://github.com/dtr-org/unit-e/pull/1099/commits/b54e3a7b54209edc07c7b501db127bc04519c9b9 ) and I am adding a test for it.

Reply:
Added the test in https://github.com/dtr-org/unit-e/pull/1099/commits/469d85440b408f49e3bb52e09af1150c78a9c9e7

Note that although `BlockValidator` is not used in `LegacyValidationInterface` yet, the `CheckCoinbaseTransaction` function is already invoked from `ConnectBlock` before checking the stake.